pavelib/prereqs.py

```diff
@@ -195,22 +195,12 @@ def uninstall_python_packages():
     # to really really get rid of it.
     for _ in range(3):
         uninstalled = False
-        frozen = sh("pip freeze", capture=True).splitlines()
-
-        # Uninstall South
-        if any(line.startswith("South") for line in frozen):
-            sh("pip uninstall --disable-pip-version-check -y South")
-            uninstalled = True
-
-        # Uninstall edx-val
-        if any("edxval" in line for line in frozen):
-            sh("pip uninstall --disable-pip-version-check -y edxval")
-            uninstalled = True
-
-        # Uninstall django-storages
-        if any("django-storages==" in line for line in frozen):
-            sh("pip uninstall --disable-pip-version-check -y django-storages")
-            uninstalled = True
+        frozen = sh("pip freeze", capture=True)
+
+        for package in ("South", "edxval", "django-storages", "django-oauth2-provider"):
+            if "{}==".format(package) in frozen:
```

Comment:
In theory we don't need to uninstall the other packages any more.  I'm not a fan of getting rid of them just because we borked the code, though.

Also, you need to change the expected_version from 2 to 3 so that the uninstall will run at all.